dll/win32/shell32/CDefViewBckgrndMenu.cpp

```diff
@@ -199,6 +199,18 @@ CDefViewBckgrndMenu::QueryContextMenu(HMENU hMenu, UINT indexMenu, UINT idCmdFir
             EnableMenuItem(hMenuPart, FCIDM_SHVIEW_INSERTLINK, MF_BYCOMMAND | MF_GRAYED);
         }
 
+        /* Disable Auto Arrange when in details or list view */
+        CComPtr<IFolderView> pfv;
+        UINT viewMode = 0;
+        if (SUCCEEDED(IUnknown_QueryService(m_site, SID_IFolderView, IID_PPV_ARG(IFolderView, &pfv))))
+        {
+            if  (SUCCEEDED(pfv->GetCurrentViewMode(&viewMode)))
+            {
+                if (viewMode == FVM_DETAILS || viewMode == FVM_LIST)
```

Comment:
There is no "else" case here, contrary to above?

Reply:
The menu resource is loaded every time the context menu is launched in CDefViewBckgrndMenu.cpp, so it only needs to be disabled if needed. Shellbrowser reuses the same  menu, so it needs to be enabled or disabled every time the menu is displayed.